apiserver/facades/client/application/deployrepository.go

```diff
@@ -93,10 +93,10 @@ func NewDeployFromRepositoryAPI(
 func (api *DeployFromRepositoryAPI) DeployFromRepository(ctx context.Context, arg params.DeployFromRepositoryArg) (params.DeployFromRepositoryInfo, []*params.PendingResourceUpload, []error) {
 	api.logger.Tracef("deployOneFromRepository(%s)", pretty.Sprint(arg))
 	// Validate the args.
-	dt, addPendingResourceErrs := api.validator.ValidateArg(ctx, arg)
+	dt, errs := api.validator.ValidateArg(ctx, arg)
```

Comment:
Not for this PR, but this validate function is very misnamed. It does a lot more than validate. Seems like it should be changed to `generateDeployTemplate` or something of that sort. It even calls out to charmhub.